angular-ngrx-scss/src/app/auth/services/auth.guard.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthGuard implements CanActivate {
+  constructor(private authService: AuthService, private router: Router) {}
+
+  canActivate():
+    | Observable<boolean | UrlTree>
+    | Promise<boolean | UrlTree>
+    | boolean
+    | UrlTree {
+    if (this.authService.isAuthenticated()) {
+      return true;
+    }
+    return this.router.createUrlTree(['/signin']);
```

Comment:
just for reference this could be simplified to
```suggestion
    return this.authService.isAuthenticated() || this.router.createUrlTree(['/signin']);
```

But not necessary to change if you prefer the more verbose approach